homeassistant/components/elkm1/__init__.py

```diff
@@ -521,3 +505,34 @@ def device_info(self) -> DeviceInfo:
         if self._mac:
             device_info[ATTR_CONNECTIONS] = {(CONNECTION_NETWORK_MAC, self._mac)}
         return device_info
+
+
+_T = TypeVar("_T", bound=ElkEntity)
+
+
+def create_elk_entities(
+    elk_data: dict[str, Any],
+    elk_elements: list[Element],
+    element_type: str,
+    class_: Any,
+    entities: list[_T],
+) -> list[_T] | None:
```

Comment:
Will this signature work instead?

```
def create_elk_entities(
    elk_data: dict[str, Any],
    elk_elements: list[Element],
    element_type: str,
    class_: Any,
    entities: list[ElkEntity],
) -> list[ElkEntity] | None:
```

Reply:
When the `TypeVar` is not used (i.e.: change the signature as you mention above) then in `switch.py` (and others places) I get the following error from `mypy`:
`Argument 5 to "create_elk_entities" has incompatible type "List[ElkOutput]"; expected "List[ElkEntity]"  [arg-type]`

Also note that I had to move `create_elk_entities` down after `ElkEntities` as `TypeVar` would not accept a forward reference.

Reply:
You could make the type `ElkEntity | ElkAttachedEntity`

Reply:
Same(ish). `Argument 5 to "create_elk_entities" has incompatible type "List[ElkOutput]"; expected "List[Union[ElkEntity, ElkAttachedEntity]]"  [arg-type]`

Reply:
https://github.com/bdraco/home-assistant/commit/c9f246ab046045be3a750e75f94f1184be0c9e38

Reply:
I simplified your suggestion, hope you like it. What about `TypeVar` is it that we're avoiding it? I saw it throughout the code base.

Reply:
> I simplified your suggestion, hope you like it.

Thats cleaner 👍

Reply:
> What about `TypeVar` is it that we're avoiding it? I saw it throughout the code base.

It introduces complexity that we can avoid here.